### PR TITLE
[PWX-29160] Upgrade azure-sdk-for-go clients api to version '2021-07-01'

### PR DESCRIPTION
**Pull request by alicelyu-px**
**What this PR does / why we need it**:
Upgrade azure-sdk-for-go client api to version '2021-07-01`
This is needed to support no-down-time resize for azure clusters.

**Special notes for your reviewer**:
Tried to run BVT/FVT following wiki 
https://portworx.atlassian.net/wiki/spaces/PE/pages/440631299/Running+BVT+FVT+on+branches
Build failed for unrelated failures, see https://jenkins.pwx.dev.purestorage.com/job/Dev/job/Porx-02/196/

travis build for porx with this branch succeeded https://github.com/portworx/porx/commit/c9dfb809e5ec977c3acd4081eb54697201395e7c

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -32,8 +31,72 @@ require (
 	k8s.io/klog v1.0.0
 )
 
+require (
```
Comment (nrevanna) [inline go.mod:34]: I think we should have a single require section and not two.
@dahuang-purestorage ?
Also, do you see any concerns with these many `require` additions?

**Review by nrevanna — COMMENTED** [review]
- Azure code changes look good to me
- Please wait in Daniel's comments for go.mod changes.

**Review by nrevanna — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (dahuang-purestorage) [reply go.mod:34]: this is fine. that's a new change in 1.18 where they separate require for direct and indirect dependencies.

**Review by dahuang-purestorage — APPROVED** [review]
LGTM on the go mod side

**Pull request merged**